WordPress/Classes/Services/SharingSyncService.swift

```diff
@@ -28,18 +28,27 @@ import WordPressKit
     ///     - failure: An optional failure block accepting an `NSError` parameter.
     ///
     @objc open func syncPublicizeConnectionsForBlog(_ blog: Blog, success: (() -> Void)?, failure: ((NSError?) -> Void)?) {
-        let blogObjectID = blog.objectID
-        guard let remote = remoteForBlog(blog) else {
+        guard let remote = remoteForBlog(blog),
+              let blogID = blog.dotComID else {
             failure?(Error.siteWithNoRemote as NSError)
             return
         }
 
-        remote.getPublicizeConnections(blog.dotComID!, success: { remoteConnections in
+        remote.getPublicizeConnections(blogID, success: { [blogObjectID = blog.objectID] remoteConnections in
+            let currentUserID: NSNumber = self.coreDataStack.performQuery { context in
+                guard let blog = try? context.existingObject(with: blogObjectID) as? Blog,
+                      let accountID = blog.account?.userID else {
+                    return NSNumber(value: 0)
+                }
+                return accountID
+            }
+
+            // Ensure that we're only processing shared or owned connections
+            let authorizedConnections = remoteConnections.filter { $0.shared || $0.userID.isEqual(to: currentUserID) }
```

Comment:
No issue with the code, just a comment. It's strange to me that we're having the client filter out this info instead of doing it on the backend.

Reply:
Yeah, that is true. There could be a special purpose behind why it's done this way, or if it's indeed a mistake, I'll look into opening a ticket for the backend side.